### PR TITLE
Pack malloc to new

### PR DESCRIPTION
**Pull request by jmah76**
#### Description
Changed all the C style memory allocation in vpr/pack to C++ style memory allocation. 

#### Checklist:

- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- X ] All new and existing tests passed

### CODE REVIEW

**Review by hzeller — COMMENTED** [review]
Some initial pass. Mostly

  * Don't use parenthesis around the `delete` calls, as it might look like a function call, obscuring the language feature
  * various raw arrays (the ones that require `delete[]` might probably be worthwhile to be considered to be converted to containers (possibly in separate PRs)

File: libs/libarchfpga/src/arch_util.cpp [hunk 1]
```diff
@@ -316,78 +316,46 @@ static void free_pb_graph(t_pb_graph_node* pb_graph_node) {
...
-                vtr::free(pb_graph_node->input_pins[i][j].pin_timing_del_max);
-            if (pb_graph_node->input_pins[i][j].pin_timing_del_min)
-                vtr::free(pb_graph_node->input_pins[i][j].pin_timing_del_min);
-            if (pb_graph_node->input_pins[i][j].input_edges)
-                vtr::free(pb_graph_node->input_pins[i][j].input_edges);
-            if (pb_graph_node->input_pins[i][j].output_edges)
-                vtr::free(pb_graph_node->input_pins[i][j].output_edges);
             if (pb_graph_node->input_pins[i][j].parent_pin_class)
-                vtr::free(pb_graph_node->input_pins[i][j].parent_pin_class);
+                delete[](pb_graph_node->input_pins[i][j].parent_pin_class);
```
[the remark this answers is not in the document]
Comment (vaughnbetz) [reply libs/libarchfpga/src/arch_util.cpp]: Would be a good change but I'd like to get rid of our mixed use of free/delete/new/malloc (dangerous if the underlying new doesn't call malloc etc.) So I prefer to change them all now; Jennifer please file an issue to track changing specific variables to arrays / vectors.

File: vpr/src/pack/cluster.cpp [hunk 2]
```diff
@@ -189,8 +189,9 @@ std::map<t_logical_block_type_ptr, size_t> do_clustering(const t_packer_opts& pa
     get_max_cluster_size_and_pb_depth(helper_ctx.max_cluster_size, max_pb_depth);
 
     if (packer_opts.hill_climbing_flag) {
-        clustering_data.hill_climbing_inputs_avail = (int*)vtr::calloc(helper_ctx.max_cluster_size + 1,
-                                                                       sizeof(int));
+        clustering_data.hill_climbing_inputs_avail = new int[helper_ctx.max_cluster_size + 1];
```
[the remark this answers is not in the document]
Comment (vaughnbetz) [reply vpr/src/pack/cluster.cpp:192]: Should be doable; another issue to file Jennifer since I think we should land this PR first.

**Comment by vaughnbetz** [comment]
CI failures seem to be due to machines not available; try rerunning failed tests once the others complete.

File: vpr/src/pack/cluster.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmah76) [reply vpr/src/pack/cluster.cpp:192]: I'll be honest, I don't know why + 1.

**Comment by jmah76** [comment]
@vaughnbetz Reran the failed tests and they've all passed. I think this is ready to merge.

**Comment by vaughnbetz** [comment]
Same (paranoid) thing to test: do a QoR run and attach results.  OK if that run includes both sets of changes.

**Comment by jmah76** [comment]
Full QoR results are on my wiki page.
image

**Pull request merged**